src/Verlite.Core/VersionCalculator.cs

```diff
@@ -51,6 +54,7 @@ public static SemVer NextVersion(SemVer lastTag, VersionCalculationOptions optio
 			else
 				return options.AutoIncrement switch
 				{
+					VersionPart.None => lastTag,
```

Comment:
Rather than trying to shoehorn it into the do-the-bump part, I think it'd be better to handle the condition prior:

```patch
		public static SemVer NextVersion(SemVer lastTag, VersionCalculationOptions options)
		{
			if (options.MinimumVersion > lastTag.CoreVersion)
				return options.MinimumVersion;
+			else if (options.AutoIncrement == VersionPart.None)
+				return lastTag;
			else if (lastTag.Prerelease is not null)
			{
				var ret = lastTag;
				ret.BuildMetadata = null;
				return ret;
			}
			else
				return options.AutoIncrement switch
				{
					VersionPart.Patch => new SemVer(lastTag.Major, lastTag.Minor, lastTag.Patch + 1),
					VersionPart.Minor => new SemVer(lastTag.Major, lastTag.Minor + 1, 0),
					VersionPart.Major => new SemVer(lastTag.Major + 1, 0, 0),
					_ => throw new InvalidOperationException("NextVersion(): Can only bump by major, minor, or patch (default)."),
				};
		}
```

This would then leave the metadata untouched too, as your unit test implies. Assuming the test expresses the correct behavior?